packages/state-transition/src/epoch/processPendingDeposits.ts

```diff
@@ -0,0 +1,117 @@
+import {FAR_FUTURE_EPOCH, ForkSeq, GENESIS_SLOT, MAX_PENDING_DEPOSITS_PER_EPOCH} from "@lodestar/params";
+import {PendingDeposit} from "@lodestar/types/lib/electra/types.js";
+import {CachedBeaconStateElectra, EpochTransitionCache} from "../types.js";
+import {increaseBalance} from "../util/balance.js";
+import {getActivationExitChurnLimit} from "../util/validator.js";
+import {computeStartSlotAtEpoch} from "../util/epoch.js";
+import {addValidatorToRegistry, isValidDepositSignature} from "../block/processDeposit.js";
+
+/**
+ * Starting from Electra:
+ * Process pending balance deposits from state subject to churn limit and depsoitBalanceToConsume.
+ * For each eligible `deposit`, call `increaseBalance()`.
+ * Remove the processed deposits from `state.pendingDeposits`.
+ * Update `state.depositBalanceToConsume` for the next epoch
+ *
+ * TODO Electra: Update ssz library to support batch push to `pendingDeposits`
+ */
+export function processPendingDeposits(state: CachedBeaconStateElectra, cache: EpochTransitionCache): void {
+  const nextEpoch = state.epochCtx.epoch + 1;
+  const availableForProcessing = state.depositBalanceToConsume + BigInt(getActivationExitChurnLimit(state.epochCtx));
+  let processedAmount = 0;
+  let nextDepositIndex = 0;
+  const depositsToPostpone = [];
+  let isChurnLimitReached = false;
+  const finalizedSlot = computeStartSlotAtEpoch(state.finalizedCheckpoint.epoch);
+
+  for (const deposit of state.pendingDeposits.getAllReadonly()) {
+    // Do not process deposit requests if Eth1 bridge deposits are not yet applied.
+    if (
+      // Is deposit request
+      deposit.slot > GENESIS_SLOT &&
+      // There are pending Eth1 bridge deposits
+      state.eth1DepositIndex < state.depositRequestsStartIndex
+    ) {
+      break;
+    }
+
+    // Check if deposit has been finalized, otherwise, stop processing.
+    if (deposit.slot > finalizedSlot) {
```

Comment:
shouldn't this be

```suggestion
    if (deposit.slot >= finalizedSlot) {
```

Reply:
`> finalizedSlot` means "not finalized" so it makes sense